src/js/relayclient/RelayClient.js

```diff
@@ -416,11 +416,19 @@ class RelayClient {
     fixTransactionReceiptResp(resp) {
         if (resp && resp.result && resp.result.logs) {
             let logs = abi_decoder.decodeLogs(resp.result.logs);
-            let relayed = logs.find(e => e && e.name == 'TransactionRelayed');
-            if (relayed && relayed.events.find(e => e.name == "status").value != 0) {
-                console.log("reverted relayed transaction. changing status to zero");
+            let canRelayFailed = logs.find(e => e && e.name == 'CanRelayFailed');
+            let transactionRelayed = logs.find(e => e && e.name == 'TransactionRelayed');
+
+            const setErrorStatus = (reason) => {
+                console.log(`${reason}. changing status to zero`)
                 resp.result.status = 0
             }
+
+            if (canRelayFailed) {
+                setErrorStatus(`canRelay failed: ${canRelayFailed.find(e => e.name == "reason").value}`)
+            } else if (transactionRelayed && transactionRelayed.events.find(e => e.name == "status").value != 0) {
+                setErrorStatus("reverted relayed transaction")
```

Comment:
any special reason to hide the status from the client?

Reply:
Note that the original message didn't display any status codes: I attempted to leave the client as unchanged as possible. I can of course add this value though, as I did in the other clause of the `if` statement.